codesigndocuitests/codesigngroup.go

```diff
@@ -0,0 +1,409 @@
+package codesigndocuitests
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/goinp/goinp"
+	"github.com/bitrise-tools/codesigndoc/codesign"
+	"github.com/bitrise-tools/go-xcode/certificateutil"
+	"github.com/bitrise-tools/go-xcode/export"
+	"github.com/bitrise-tools/go-xcode/exportoptions"
+	"github.com/bitrise-tools/go-xcode/plistutil"
+	"github.com/bitrise-tools/go-xcode/profileutil"
+)
+
+// extractCertificatesAndProfiles returns the certificates and provisioning profiles of the given codesign group
+func extractCertificatesAndProfiles(codeSignGroups ...export.CodeSignGroup) ([]certificateutil.CertificateInfoModel, []profileutil.ProvisioningProfileInfoModel) {
+	certificateMap := map[string]certificateutil.CertificateInfoModel{}
+	profilesMap := map[string]profileutil.ProvisioningProfileInfoModel{}
+	for _, group := range codeSignGroups {
+		certificate := group.Certificate()
+		certificateMap[certificate.Serial] = certificate
+
+		for _, profile := range group.BundleIDProfileMap() {
+			profilesMap[profile.UUID] = profile
+		}
+	}
+
+	certificates := []certificateutil.CertificateInfoModel{}
+	profiles := []profileutil.ProvisioningProfileInfoModel{}
+	for _, certificate := range certificateMap {
+		certificates = append(certificates, certificate)
+	}
+	for _, profile := range profilesMap {
+		profiles = append(profiles, profile)
+	}
+	return certificates, profiles
+}
+
+// codesignMethod returns which code sign method type is allowed by the given codesign group
+func codesignMethod(group export.CodeSignGroup) string {
+	for _, profile := range group.BundleIDProfileMap() {
+		return string(profile.ExportType)
+	}
+	return ""
+}
+
+// printCodesignGroup prints the given codesign group
+func printCodesignGroup(group export.CodeSignGroup) {
+	fmt.Printf("%s %s (%s)\n", colorstring.Green("development team:"), group.Certificate().TeamName, group.Certificate().TeamID)
+	fmt.Printf("%s %s [%s]\n", colorstring.Green("codesign identity:"), group.Certificate().CommonName, group.Certificate().Serial)
+
+	idx := -1
+	for bundleID, profile := range group.BundleIDProfileMap() {
+		idx++
+		if idx == 0 {
+			fmt.Printf("%s %s -> %s\n", colorstring.Greenf("provisioning profiles:"), profile.Name, bundleID)
+		} else {
+			fmt.Printf("%s%s -> %s\n", strings.Repeat(" ", len("provisioning profiles: ")), profile.Name, bundleID)
+		}
+	}
+}
+
+// collectExportCertificate returns the certificate to use for the UITest-Runner
+func collectExportCertificate(installedCertificates []certificateutil.CertificateInfoModel) ([]certificateutil.CertificateInfoModel, error) {
+	var selectedCertificates []certificateutil.CertificateInfoModel
+
+	// Codesign method
+	codesignMethods := []string{"development", "app-store", "ad-hoc", "enterprise"}
```

Comment:
Can this be a const somewhere?

Reply:
Golang doesn't allow us to create an array / map as a const:
`const initializer []string literal is not a constant`

https://groups.google.com/forum/#!topic/golang-nuts/5aynucvg96I